src/payment/checkout/payment-form/flex-microform/FlexMicroform.jsx

```diff
@@ -25,26 +23,16 @@ class FlexMicroform extends React.Component {
     this.initialize();
   }
 
-  componentWillUnmount() {
-    if (this.timer) {
-      clearTimeout(this.timer);
-      this.timer = null;
-    }
-  }
-
   initialize = () => {
     if (window.microform !== null) {
       return;
     }
-    if (!this.props.captureKeyId) {
-      return;
-    }
-    if (this.timer) {
-      clearTimeout(this.timer);
-      this.timer = null;
-    }
-    if (typeof window.Flex === 'undefined') {
-      this.timer = setTimeout(this.initialize, POLLING_TIME_MS);
+    if (!this.props.captureKeyId || typeof window.Flex === 'undefined') {
```

Comment:
when the page first initializes it is expected that `this.props.captureKeyId` is `null` it will be non-null once the AJAX call to the backend completes.

Reply:
ah good point